### PR TITLE
Add event on object deletion

### PR DESCRIPTION
**Pull request by 2mt-heuser**
Adds a **'pages.object.delete'** event on single and mass-deletion of objects.

See #402

### CODE REVIEW

File: controllers/Index.php [hunk 1]
```diff
@@ -175,8 +175,15 @@ public function onDelete()
...
         $type = Request::input('objectType');
+        $object = $this->loadObject($type, trim(Request::input('objectPath')));
 
-        $deletedObjects = $this->loadObject($type, trim(Request::input('objectPath')))->delete();
+        $deletedObjects = $object->delete();
+
+        /*
+         * Extensibility
+         */
+        Event::fire('pages.object.delete', [$this, $object, $type]);
```
Comment (bennothommo) [inline controllers/Index.php]: @2mt-heuser Would you mind adding a comment explaining the event and its arguments? Similar to this:
https://github.com/octobercms/october/blob/315e2dcac57741df66efa07a001f08730e7bc943/modules/cms/classes/CmsObject.php#L140

Comment (2mt-heuser) [reply controllers/Index.php]: Added comments to the two new delete event fires as well as the save one that only had "extensibility" in there.

**Comment by bennothommo** [comment]
Thanks for that @2mt-heuser - looks great. I've just made some slight adjustments to the grammar and formatting of the event docblock. Otherwise, all that's left is for this to be tested. :)

File: controllers/Index.php [hunk 2]
```diff
@@ -211,6 +255,15 @@ public function onDeleteObjects()
                 }
 
                 $deletedObjects = $object->delete();
+
+                /*
+                * @event pages.object.delete
```
Comment (LukeTowers) [inline controllers/Index.php]: `@event` comment should only exist for one instance of the event being fired, it's used by API documentation generators so there is no reason to have more than one definition for a single event

Comment (2mt-heuser) [reply controllers/Index.php]: Will remove this one then.

**Comment by vinnchan** [comment]
Hi, any updates on this feature request, because I do need this event to delete Rainlab static page in database. Recommendation for alternatives are welcome as well, thank you for your times

**Comment by 2mt-heuser** [comment]
> Hi, any updates on this feature request, because I do need this event to delete Rainlab static page in database. Recommendation for alternatives are welcome as well, thank you for your times

Somehow it never got merged although approved.

**Comment by vinnchan** [comment]
> > Hi, any updates on this feature request, because I do need this event to delete Rainlab static page in database. Recommendation for alternatives are welcome as well, thank you for your times
> 
> Somehow it never got merged although approved.

I guess this will release an update right ? Thanks

**Comment by LukeTowers** [comment]
Technically both of these events are redundant, but for BC we're not going to remove the .save event.

I'm going to close this unless someone can provide a good reason why the existing `model.afterDelete` event isn't good enough to meet their needs: https://octobercms.com/docs/api/model/afterdelete

```php
\RainLab\Pages\Classes\Page::extend(function ($model) {
    $model->bindEvent('model.afterDelete', function () use ($model) {
        // do stuff
    });
});
```

**Pull request closed without merging**